slsim/lens_pop.py

```diff
@@ -237,6 +271,36 @@ def __init__(
                     list_type="list",
                     light_profile=source_light_profile,
                 )
+            else:
+                from slsim.Sources.SupernovaeCatalog.supernovae_sample import (
```

Comment:
these lines of code following here are very similar to lines of code higher up. Perhaps we can avoid duplications here a bit as this class initialization is rather long